SECURITY.md

```diff
@@ -256,6 +256,14 @@ embargo policy to share binary distributions of the security fixes before the pu
 This includes, but is not limited to, Envoy binaries and Docker images. It is expected that
 distributors have a method to stage and validate new binaries without exposing them publicly.
 
+If Envoy is running in a hosted environment, the fix may rollout ahead of the embargo lift date
+only if the end user **cannot** access the running Envoy process and possibly reverse engineer the fix.
+If a security fix includes changes to the controlplane then the controlplane binaries, Docker images,
+and running process should not be accessible to the end user. If the above conditions hold in a hosted
+controlplane then the rollout may proceed ahead of the embargo lift date. If the controlplane produces
+XDS output containing textual keys pertaining to the fix, such as specific filter metadata keys then
```

Comment:
Nit: xDS

Reply:
xDS case is interesting as users can look under the hood and inspect Envoy configuration (in sidecar proxy cases) even for hosted environments. I'm assuming if the CVE is related to Envoy then the patch has to include a delta in Envoy config i.e. inspecting Envoy config between old and new gives away the fact that CVE fix has been rolled out, is that correct?

Reply:
I think its contextual. Just changing xDS doesn't inherently give away a vulnerability - the control plane is presumably constantly evolving. If the change is inserting a lua filter with a name "mitigate-cve-caused-by-header-foobaring" that is obviously pretty bad. If its just flipping some boolean flag, that happens to mitigate a vulnerability, its probably not giving anything away?

Reply:
This argument also holds true for non-hosted vendors who routinely make new releases and can obfuscate a CVE fix with other changes. Can they also release ahead of embargo in that case?

Reply:
As long as the end user cannot access the image, or the running program they can release early. How would a non hosted vendor do that ?

Reply:
Your point is valid for control plane. I thought this modification also covers releasing updated Envoy before the embargo. Is that for non-sidecar cases only in hosted environments where you cannot inspect the underlying pod?

Reply:
It does if the gateway is managed and otherwise not accessible to the end user including config dumps.

Reply:
Right, this is all about whether the user has access to binaries and/or other contextual signals that would allow reverse engineering the fix. For binaries, this is basically always the case. For hosted cases where binary access is not available it's contextual and I think per other threads we can leave it up to the EST.